lms/djangoapps/support/static/support/jsx/single_support_form.jsx

```diff
@@ -0,0 +1,211 @@
+/* global gettext */
+/* eslint one-var: ["error", "always"] */
+/* eslint no-alert: "error" */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import FileUpload from './file_upload';
+import ShowErrors from './errors_list';
+import LoggedInUser from './logged_in_user';
+import LoggedOutUser from './logged_out_user';
+
+// TODO
+// edx zendesk APIs
+// access token
+// custom fields ids
+// https://openedx.atlassian.net/browse/LEARNER-2736
+// https://openedx.atlassian.net/browse/LEARNER-2735
+
+class RenderForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentRequest: null,
+      errorList: [],
+    };
+    this.submitForm = this.submitForm.bind(this);
+    this.setErrorState = this.setErrorState.bind(this);
+  }
+
+  setErrorState(errors) {
+    this.setState({
+      errorList: errors,
+    });
+  }
+
+  submitForm() {
+    const url = 'https://arbisoft.zendesk.com/api/v2/tickets.json',
+      $userInfo = $('.user-info'),
+      request = new XMLHttpRequest(),
+      $course = $('#course'),
+      accessToken = 'd6ed06821334b6584dd9607d04007c281007324ed07e087879c9c44835c684da',
+      data = {
+        subject: $('#subject').val(),
+        comment: {
+          body: $('#message').val(),
+          uploads: $.map($('.uploaded-files a'), n => n.id),
+        },
+      };
+
+    let course;
+
+    if ($userInfo.length) {
+      data.requester = $userInfo.data('email');
+      course = $course.find(':selected').text();
+      if (!course.length) {
+        course = $course.val();
+      }
+    } else {
+      data.requester = $('#email').val();
+      course = $course.val();
+    }
+
+    data.custom_fields = [{
+      id: '114099484092',
+      value: course,
+    }];
+
+    if (this.validateData(data)) {
+      request.open('POST', url, true);
+      request.setRequestHeader('Authorization', `Bearer ${accessToken}`);
+      request.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+
+      request.send(JSON.stringify({
+        ticket: data,
+      }));
+
+      request.onreadystatechange = function success() {
+        if (request.readyState === 4 && request.status === 201) {
+          // TODO needs to remove after implementing success page
+          const alert = 'Request submitted successfully.';
```

Comment:
@tasawernawaz Is this what the learner sees on successfully submitting the request? On the sandbox I'm seeing what might be a slightly different version where the capitalization is different. 
![supportformrequestsuccess](https://user-images.githubusercontent.com/9041399/31912311-71917386-b811-11e7-8416-4158a0699ead.png)

Could we have a more user-friendly message here?
"Thank you for submitting a support request. Someone from our Support team will contact you soon."

@dhixonedx is this OK?

Reply:
@catong NO. This will be replace in https://openedx.atlassian.net/browse/LEARNER-2735

Reply:
OK! Let's remember to fix it - at the least, the message should be a complete sentence.

Reply:
@catong Yes we have already a story which will replace this alert.